cli/lib/snapshots.ts

```diff
@@ -17,25 +10,23 @@ export async function getSnapshotsFromAppdata(
 	snapshots = snapshots.filter( ( snapshot ) => snapshot.userId === userId );
 
 	if ( siteFolder ) {
-		const siteId = await getSiteIdFromFolder( siteFolder );
-		snapshots = snapshots.filter( ( snapshot ) => snapshot.localSiteId === siteId );
+		const site = await getSiteFromFolder( siteFolder );
+		snapshots = snapshots.filter( ( snapshot ) => snapshot.localSiteId === site.id );
 	}
 
 	return snapshots;
 }
 
-export async function addPreviewSiteToAppdata(
-	previewUrl: string,
+// Insert or update a snapshot entry (matching on atomicSiteId) in the appdata file.
+export async function upsertPreviewSiteInAppdata(
+	siteFolder: string,
 	atomicSiteId: number,
-	siteFolder: string
```

Comment:
Reordering the arguments wasn't necessary, it's just something I found intuitive. `siteFolder` comes first because it defines the site we're working with (and the function throws if this value doesn't map to an existing site). `atomicSiteId` comes second because that's the identifier used to update existing snapshots. `previewUrl` comes last because it doesn't have any side effects.